client/app/bundles/course/level/operations.ts

```diff
@@ -3,64 +3,41 @@ import { Operation } from 'store';
 import CourseAPI from 'api/course';
 import { setNotification } from 'lib/actions';
 
-import actionTypes from './constants';
+import { actions } from './store';
 
 export function fetchLevels(): Operation {
   return async (dispatch) => {
-    dispatch({ type: actionTypes.LOAD_LEVELS_REQUEST });
-    return CourseAPI.level
-      .fetch()
-      .then((response) => {
-        dispatch({
-          type: actionTypes.LOAD_LEVELS_SUCCESS,
-          levelData: response.data,
-        });
-      })
-      .catch(() => {
-        dispatch({ type: actionTypes.LOAD_LEVELS_FAILURE });
-      });
-  };
-}
-
-export function updateExpThreshold(levelNumber, newValue): Operation {
-  return async (dispatch) => {
-    dispatch({
-      type: actionTypes.UPDATE_EXP_THRESHOLD,
-      payload: { levelNumber, newValue },
+    return CourseAPI.level.fetch().then((response) => {
+      dispatch(
+        actions.saveLevelsData({
+          levels: response.data.levels,
+          canManage: response.data.canManage,
+        }),
+      );
     });
   };
 }
 
-export function sortLevels(): Operation {
-  return async (dispatch) => {
-    dispatch({ type: actionTypes.SORT_LEVELS });
-  };
-}
-
-export function addLevel(): Operation {
-  return async (dispatch) => {
-    dispatch({ type: actionTypes.ADD_LEVEL });
-  };
-}
-
-export function deleteLevel(levelNumber): Operation {
-  return async (dispatch) => {
-    dispatch({ type: actionTypes.DELETE_LEVEL, payload: { levelNumber } });
-  };
-}
-
-export function saveLevels(levels, successMessage, failureMessage): Operation {
+export function saveLevels(
+  levels: number[],
+  successMessage,
+  failureMessage,
+): Operation {
+  levels.sort((a, b) => a - b);
```

Comment:
1. It's better to assign a new variable `const sortedLevels` instead of using the modified `levels` and using it below. It's not quite obvious when reading this function, on whether the `level` called below has been sorted or not.

Something like
```
return CourseAPI.level
      .save(sortedLevels)
```

2. Don't use variable names a and b in here; it looks ugly. Use l1 and l2 instead (or, even better, `level1` and `level2`

Reply:
Done